### PR TITLE
build: update golang k8s API's to v0.23

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Comment by nbalacha** [comment]
Please explain that this is upgrading to k8s v1.23 api in the PR title and commit msg

File: go.mod [hunk 1]
```diff
@@ -3,22 +3,22 @@ module github.com/red-hat-storage/lvm-operator
 go 1.17
 
 require (
-	github.com/go-logr/logr v0.4.0
+	github.com/go-logr/logr v1.2.0
```
Comment (leelavg) [inline go.mod:6]: - seeing CI pass is a good sign however pls do test locally once
- iirc some combo of logr and controller-runtime failed during the initial stages of the project

Comment (nbalacha) [reply go.mod:6]: @iamniting , I will merge this once you confirm that the local tests pass.

**Pull request merged**